packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -98,13 +99,24 @@ function InsertionPointInserter( {
 
 function InsertionPointPopover( {
 	clientId,
+	rootClientId,
 	isInserterShown,
 	isInserterForced,
 	setIsInserterForced,
 	containerRef,
 	showInsertionPoint,
 } ) {
-	const element = getBlockDOMNode( clientId );
+	const appenderNodesMap = useContext( AppenderNodesContext );
+	const element = useMemo( () => {
+		if ( clientId ) {
+			return getBlockDOMNode( clientId );
+		}
+
+		// Can't find the element, might be at the end of the block list, or inside an empty block list.
+		// We instead try to find the "Appender" and place the indicator above it.
+		// `rootClientId` could be null or undefined when there's no parent block, we normalize it to an empty string.
+		return appenderNodesMap.get( rootClientId || '' );
```

Comment:
The insertion point is added at the root of the whole tree of blocks, so I think `rootClientId` will always be `''`, never an id?

Reply:
In other words, does the indicator work for nested appenders?

Reply:
It could be an id if it's in the Widget Editor though, `widget-area` is kind of like a special case for this 😜.

Also, `<InsertionPoint>` only handles _getting_ of the appender nodes. `<BlockList>` is where _setting_ happens, which can be in `<InnerBlocks>` or `widget-area`. What I'm trying to say here is that a `<InsertionPoint>` could have multiple `<BlockList>` hence could also have multiple `<BlockListAppender>`.

Reply:
> In other words, does the indicator work for nested appenders?

It should, at least for my testing below.

![image](https://user-images.githubusercontent.com/7753001/95294925-b3436b00-08a8-11eb-8289-781d73a05e73.png)